### ISSUE TITLE
Callback never called

### ISSUE THREAD

**Issue by deakjahn** [MaxMotovilov/eeMD#1]
I'd love to but the callback is never called, it seems:

    require(['https://cdnjs.cloudflare.com/ajax/libs/three.js/110/three.min'], function(three) {
      window.THREE = three;
      ...

I have it as simple as this. I check the url passed to `provide(deps)` and the URL is correct. The browser dev tool shows the JS itself is loaded but the `cb.apply()` is never executed.

**Comment by MaxMotovilov** [comment]
See https://github.com/MaxMotovilov/eeMD#limitations - specifically, "all modules must be named". The code is so short because, among other things, it doesn't parse the module IDs out of URLs and, more importantly, doesn't track modules as they load.

The callback would fire if you'd replace `define(["exports"], Aa)` with `define("three.min", ["exports"], Aa)` in the body of your UMD module and use:

```Javascript
require
  .config({paths: {"": "https://cdnjs.cloudflare.com/ajax/libs/three.js/110/"}})
  (["three.min"], function(three) {
    window.THREE = three;
    ...
  });
```
to load it. Unfortunately, `exports` is not a special dependency eeMD recognizes so it still won't fire unless you actually load a module with `define("exports", ....)` in its body with a separate `<script>` statement. Which is, obviously, not what you wanted here.

Looks like eeMD is too limited/specialized for your use case :-( That said, adding support for `exports` might prove relatively lightweight so you're welcome to submit a PR. I am rather pleasantly surprised that someone takes an interest in this little project and would be happy to help :-)

**Comment by deakjahn** [comment]
The module to load is not mine, it's the rather widespread `three.js` (https://cdnjs.cloudflare.com/ajax/libs/three.js/110/three.js in original form), coming from a CDN, so I can't really modify it. It seems to me that it's named correctly.

Actually, I've been looking hard to find a minimal AMD loader and checked out a long list until I arrived at yours. :-) Basically, I need a loader only solution that's compatible with RequireJS (not necessarily in all its config, just in function) that I can use in a scenario where RequireJS is either present in the framework or not, I can't influence it. So, I need a parallel loader, independent of that RequireJS that can do the same (I modifiy the name to `requireX()` to avoid collision). These modules all load perfectly with RequireJS and operate as needed, I just need that parallel solution.

**Comment by deakjahn** [comment]
I have more than one module to solve, this one looks a tad different, it's named in its export. Would this be the right approach?

    require.config({
      paths: {
        'pdfjs-dist/build/pdf': 'https://cdnjs.cloudflare.com/ajax/libs/pdf.js/2.3.200/pdf.min',
        'pdfjs-dist/build/pdf.worker': 'https://cdnjs.cloudflare.com/ajax/libs/pdf.js/2.3.200/pdf.worker.min',
      }
    });
    
    require(["pdfjs-dist/build/pdf", "pdfjs-dist/build/pdf.worker"], function(pdfjsLib, pdfjsWorker) {
      console.log(pdfjsLib);
      console.log(pdfjsWorker);

or

    require.config({
      paths: {
        '': 'https://cdnjs.cloudflare.com/ajax/libs/pdf.js/2.3.200/',
        'pdfjs-dist/build/pdf': 'pdf.min',
        'pdfjs-dist/build/pdf.worker': 'pdf.worker.min',
      }
    });
    
    require(["pdfjs-dist/build/pdf", "pdfjs-dist/build/pdf.worker"], function(pdfjsLib, pdfjsWorker) {
      console.log(pdfjsLib);
      console.log(pdfjsWorker);

Both try to load locally, not from the CDN url, though.

**Comment by deakjahn** [comment]
OK, with direct chaining (not specifying config separately) it works. But only if I define the URLs in full, not with the "" prefix (what isn't really a problem for me, especially as in some cases I need to load from more than one source):

    require.config({
      paths: {
        'pdfjs-dist/build/pdf': 'https://cdnjs.cloudflare.com/ajax/libs/pdf.js/2.3.200/pdf.min',
        'pdfjs-dist/build/pdf.worker': 'https://cdnjs.cloudflare.com/ajax/libs/pdf.js/2.3.200/pdf.worker.min',
      }
    })(['pdfjs-dist/build/pdf', 'pdfjs-dist/build/pdf.worker'], function(pdfjsLib, pdfjsWorker) {
      console.log(pdfjsLib);
      console.log(pdfjsWorker);

**Comment by MaxMotovilov** [comment]
Yes - since these modules have physical paths that don't contain their module IDs as suffixes, you can't very well use a common prefix in the `config.paths`. RequireJS wouldn't let you do this either.

Historically, this mini-loader came from an effort to optimize page loading time for a specific product. Every possible tradeoff was made in favor of reducing size of the code. Since I had control over content and physical paths of the bundles being loaded -- they were generated and deployed by the product's build script -- the use case of a public CDN was left by wayside.

It's been awhile since I touched this project but I vaguely remember that the decision not to support nameless modules (i.e. the two-argument form of `define()`) was fairly fundamental, as the loader needs some way to tell just what module is being asynchronously defined and the indirect means of achieving that entail quite a lot of code. The other limitations may or may not be easier to overcome; but the goals of keeping the loader tiny and making it sufficiently universal do appear mutually exclusive.

**Comment by deakjahn** [comment]
I have one main module (namely, Flutter CanvasKit) where the define is anonymous, so I opened a bug and try to convince them to add the name. :-)
